### PR TITLE
solution review

### PR DESCRIPTION
**Pull request by akhadov11**

### CODE REVIEW

File: app/create_file.py [hunk 1]
```diff
@@ -1 +1,56 @@
-# write your code here
+from datetime import datetime
+from sys import argv
+from os import mkdir, chdir
```
Comment (akhadov11) [inline app/create_file.py:3]: it's more common practice to import `os` and `sys` modules like this `import os import sys` because `os.mkdir` `orsys.argv` is much more readable then bare `argv `and `mkdir`

File: app/create_file.py [hunk 2]
```diff
@@ -1 +1,56 @@
...
+
+filename = ""
+directories = []
+
+if "-f" in argv:
```
Comment (akhadov11) [inline app/create_file.py:8]: probably it would be better to have constants for your flags ("-f", "-d") or even an enum. because if you'll need to change them someday it would be easier to change that in one place than all the hardcoded values

File: app/create_file.py [hunk 3]
```diff
@@ -1 +1,56 @@
-# write your code here
+from datetime import datetime
+from sys import argv
+from os import mkdir, chdir
+
+filename = ""
+directories = []
```
Comment (akhadov11) [inline app/create_file.py:6]: global mutable variables are always dangerous to use as they can be mutated from a lot of places

File: app/create_file.py [hunk 4]
```diff
@@ -1 +1,56 @@
...
+
+if "-f" in argv:
+    index = 0
+
+    for _ in range(len(argv) - 1):
```
Comment (akhadov11) [inline app/create_file.py:11]: underscore for index is used only when it is not used in the loop body and it's not important but you're actively using it later, so rename it to something more readable

File: app/create_file.py [hunk 5]
```diff
@@ -1 +1,56 @@
...
+        index += 1
+        if argv[_: _ + 1] == ["-f"]:
```
Comment (akhadov11) [inline app/create_file.py:13]: the line is hard to read, can be simplified by using `list.index()` method or at least by retrieving and comparing single elem, not the slice

File: app/create_file.py [hunk 6]
```diff
@@ -1 +1,56 @@
...
+            break
+
+    filename = argv[index]
+
+    if "-d" in argv:
+        index2 = 0
+
+        for _ in range(len(argv) - 1):
```
Comment (akhadov11) [inline app/create_file.py:21]: same here as above for naming

File: app/create_file.py [hunk 7]
```diff
@@ -1 +1,56 @@
...
+            index2 += 1
+            if argv[_: _ + 1] == ["-d"]:
```
Comment (akhadov11) [inline app/create_file.py:23]: same here for `index()`

File: app/create_file.py [hunk 8]
```diff
@@ -1 +1,56 @@
...
+                break
+
+        directories = argv[index2:index - 1]
+else:
+    filename = "file.txt"
+
+    if "-d" in argv:
```
Comment (akhadov11) [inline app/create_file.py:30]: can be merged with `else `above to elif `"-d" in argv`

File: app/create_file.py [hunk 9]
```diff
@@ -1 +1,56 @@
...
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index:len(argv)]
+
+file_data = [datetime.today().strftime('%y:%d:%m %H-%M-%S'), "\n"]
+
+while True:
+    line = input("Enter content line: ")
+    if line == "exit":
```
Comment (akhadov11) [inline app/create_file.py:44]: Radme says that the stop-word is "stop" not "exit"

File: app/create_file.py [hunk 10]
```diff
@@ -1 +1,56 @@
...
+        break
+    file_data.extend([line])
```
Comment (akhadov11) [inline app/create_file.py:46]: You can append the line to the end of the list using `.append()` method instead of extending with a single-element list

File: app/create_file.py [hunk 11]
```diff
@@ -1 +1,56 @@
...
+
+for directory in directories:
```
Comment (akhadov11) [inline app/create_file.py:48]: You can use `os.makedirs()` for recursive directories creation instead of yout loop

File: app/create_file.py [hunk 12]
```diff
@@ -1 +1,56 @@
...
+    try:
+        mkdir(directory)
+    except Exception:
+        pass
+    chdir(directory)
+
+with open(filename, "w") as f:
```
Comment (akhadov11) [inline app/create_file.py:55]: "w" mode means you will be rewriting the file content on each script run, for appending to the end of the file use "a" mode. also you will be creating files for each script run even if the -f flag wasn't passed, which is not the desired behavior

File: app/create_file.py [hunk 13]
```diff
@@ -1 +1,56 @@
...
+    line = input("Enter content line: ")
+    if line == "exit":
+        break
+    file_data.extend([line])
+
+for directory in directories:
+    try:
+        mkdir(directory)
+    except Exception:
+        pass
```
Comment (akhadov11) [inline app/create_file.py:52]: avoid passing broad exceptions, but anyway I suggest removing the whole loop in order to use os.makedirs

File: app/create_file.py [hunk 14]
```diff
@@ -1 +1,56 @@
...
+            index += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index:len(argv)]
+
+file_data = [datetime.today().strftime('%y:%d:%m %H-%M-%S'), "\n"]
+
+while True:
+    line = input("Enter content line: ")
```
Comment (akhadov11) [inline app/create_file.py:43]: you will be asking for input even if there wasn't -f flag passed e.g. this case `python create_file.py -d dir1 dir2`

File: app/create_file.py [hunk 15]
```diff
@@ -1 +1,56 @@
...
+
+    if "-d" in argv:
+        index = 0
+
+        for _ in range(len(argv) - 1):
+            index += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index:len(argv)]
```
Comment (akhadov11) [inline app/create_file.py:38]: can be simplified to basic slice:  `argv[index:]` - will be default grab all elements till the last one

File: app/create_file.py (hunk 10 above)
Comment (akhadov11) [reply app/create_file.py:46]: if the line is for instance "Hello", then your list looks like:  [datetime.today().strftime('%y:%d:%m %H-%M-%S'), "\n", "Hello"] - there's no "\n" newline before "Hello". so you need to append not the` line` but a line with newline defore, like this` f"\n{line}"`

**Review by akhadov11 — COMMENTED** [review]
Overall code is quite hard to read due to poor naming and high coupling. All code is in global scope without using any helper functions which makes the codestyle kind of imperative, hard to re-use, error-prone and with high complexity. I would recommend to split the code into several functions with descriptive names and use them as building blocks for this task. Each function should have it's small but well-defined responsibility so it can be easy to understand and to re-use in future. 
What's more, I found a few bugs in the code which is not really good for code-reviews: you should deliver working code for the review. Try running your code through some edge case input data and writing some tests for your code, that would alse serve as a great documentation for your code.
Also, take care about the namings - the more descriptive names you give to your variables and objects, the easier it would be for you to understand your code after returning to it in some time (or after you forget what index and index2 mean 🙂 ).
Finally, if you will import anything from this module to another one, all of your code will be run by Python as it is in global scope. So consider using some kind of main() function and calling it from an entrypoint of your script (`if __name__ == "main"`).
